Source/WebCore/Modules/streams/ReadableStream.h

```diff
@@ -190,6 +195,12 @@ class ReadableStream : public RefCounted<ReadableStream>, public ContextDestruct
     const RefPtr<InternalReadableStream> m_internalReadableStream;
 
     const RefPtr<DependencyToVisit> m_dependencyToVisit;
+    Lock m_gcLock;
+    WeakPtr<ReadableStream> m_teedBranch0ForGC WTF_GUARDED_BY_LOCK(m_gcLock);
```

Comment:
It would look safer if we were using CheckedPtr here. I know you're intending to null out the WeakPtr with a lock before destruction but we could easily introduce a bug and become thread-unsafe as long as we use a WeakPtr.